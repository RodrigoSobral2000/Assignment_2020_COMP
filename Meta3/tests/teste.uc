

int soma(void ty, void jn, void);