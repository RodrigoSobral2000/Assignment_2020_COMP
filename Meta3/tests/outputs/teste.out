Line 10, col 5: Conflicting types (got int(int,short), expected int(int,double))
Line 16, col 5: Conflicting types (got int(int,int), expected int(int,int,int))
Line 17, col 5: Conflicting types (got int(int,int), expected int(int,int,int))
Line 25, col 12: Wrong number of arguments to function teste (got 2, required 3)
Line 28, col 5: Symbol main already defined
Line 32, col 9: Conflicting types (got double, expected int)
===== Global Symbol Table =====
putchar	int(int)
getchar	int(void)
func	int(int,double)
putchar	int(int)
main	int(void)
teste	int(int,int,int)

===== Function putchar Symbol Table =====
return	int
a	char	param
b	char	param

===== Function main Symbol Table =====
return	int
i	int

===== Function teste Symbol Table =====
return	int

Program
..FuncDeclaration
....Int
....Id(func)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Double
..FuncDeclaration
....Int
....Id(putchar)
....ParamList
......ParamDeclaration
........Int
..FuncDefinition
....Int
....Id(putchar)
....ParamList
......ParamDeclaration
........Int
....FuncBody
......Return
........IntLit(5) - int
..FuncDefinition
....Int
....Id(main)
....ParamList
......ParamDeclaration
........Void
....FuncBody
......Declaration
........Int
........Id(i)
........Call - int
..........Id(func) - int(int,double)
..........IntLit(2) - int
..........IntLit(7) - int
......Return
........IntLit(0) - int
..FuncDefinition
....Int
....Id(func)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Short
....FuncBody
......Return
........IntLit(5)
..FuncDeclaration
....Int
....Id(teste)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Int
......ParamDeclaration
........Int
..FuncDeclaration
....Int
....Id(teste)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Int
..FuncDefinition
....Int
....Id(teste)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Int
....FuncBody
......Return
........IntLit(5)
..FuncDefinition
....Int
....Id(teste)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Int
......ParamDeclaration
........Int
....FuncBody
......Return
........IntLit(6) - int
..FuncDefinition
....Int
....Id(putchar)
....ParamList
......ParamDeclaration
........Char
........Id(a)
......ParamDeclaration
........Char
........Id(b)
....FuncBody
......Return
........Call - int
..........Id(teste) - int(int,int,int)
..........IntLit(5) - int
..........IntLit(5) - int
..FuncDefinition
....Int
....Id(main)
....ParamList
......ParamDeclaration
........Int
....FuncBody
......Declaration
........Int
........Id(a)
........IntLit(5) - int
......Declaration
........Short
........Id(b)
........IntLit(2) - int
......Declaration
........Double
........Id(c)
........IntLit(3) - int
......Declaration
........Int
........Id(d)
........Add - double
..........Add - int
............Id(a) - int
............Id(b) - short
..........Id(c) - double
